### PR TITLE
feat: streamer additional gov proposals

### PR DESCRIPTION
**Pull request by mtsitrin**
- [x] gov proposal to stop existing stream 
- [x] gov proposal to modify distribution weights of existing stream
- [x] gov proposal to replace distribution weights of existing stream

### CODE REVIEW

File: x/streamer/client/cli/tx_pause_stream.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		Use:   "stop-stream-proposal streamID [flags]",
+		Short: "proposal to stop an exisiting stream",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			streamID, err := strconv.ParseUint(args[0], 10, 64)
```
Comment (omritoptix) [inline x/streamer/client/cli/tx_pause_stream.go:28]: prob worth checking also if the stream exists before a proposal is sent (also applied to tx_replace and tx_update proposals)

File: x/streamer/keeper/keeper_replace_update_distribution.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		recordsMap[existingRecord.GaugeId] = existingRecord
+	}
+
+	for _, record := range records {
+		recordsMap[record.GaugeId] = record
+	}
+
+	newRecords := []types.DistrRecord{}
+	for _, val := range recordsMap {
+		if !val.Weight.Equal(sdk.ZeroInt()) {
```
Comment (omritoptix) [inline x/streamer/keeper/keeper_replace_update_distribution.go:57]: what is it's minus value? i.e weight = -3?

**Review by omritoptix — COMMENTED** [review]
1. File called tx_stop_stream. the term stop is a bit confusing as it seems like it can be started again. IMO a better term can be found e.g terminate/etc. 
2. I think  moving a stopped stream  to “finished” state can be confusing when querying a stream as it may be thought of as it was ongoing and completed. Possibly terminated/deactivated etc more suitable. related to my previous comment. 
3. What about the balance of the tokens in the streams once it’s prematurely terminated? maybe missed that part of the logic.

File: x/streamer/client/cli/tx_pauseStream.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		Use:   "stop-stream-proposal streamID [flags]",
+		Short: "proposal to stop an exisiting stream",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			streamID, err := strconv.ParseUint(args[0], 10, 64)
```
[the remark this answers is not in the document]
Comment (mtsitrin) [reply x/streamer/client/cli/tx_pauseStream.go]: it doesn't have access to the app in this stage. it's just creates the proposal message actually.
the validation logic happens in the handler

File: x/streamer/client/cli/tx_pause_stream.go (hunk 1 above)
Comment (mtsitrin) [reply x/streamer/client/cli/tx_pause_stream.go:28]: it doesn't have access to the app in this stage. it's just creates the proposal message actually.
the validation logic happens in the handler

File: x/streamer/keeper/keeper_replace_update_distribution.go (hunk 2 above)
Comment (mtsitrin) [reply x/streamer/keeper/keeper_replace_update_distribution.go:57]: it's valdiated in `distrInfo, err := types.NewDistrInfo(newRecords)`

**Comment by mtsitrin** [comment]
> File called tx_stop_stream. the term stop is a bit confusing as it seems like it can be started again. IMO a better term can 

be found e.g terminate/etc.
will change

> I think moving a stopped stream to “finished” state can be confusing when querying a stream as it may be thought of as it was ongoing and completed. Possibly terminated/deactivated etc more suitable. related to my previous comment.

I don't want to introduce additional changes in the key store for it.
u will see from the finished stream data that's it been terminated (`filled epochs < num epoch to dist`)

> What about the balance of the tokens in the streams once it’s prematurely terminated? maybe missed that part of the logic.

not sure what u mean.
the tokens for this stream are "released" for further calculations (see `TestStopStream_ModuleToDistributeCoins`)

**Review by omritoptix — COMMENTED** [review]
the cli file is still named `tx_pause_stream` probably better to change to `tx_terminate_stream` and also the `Use` and `Short` fields of the command.

**Review by omritoptix — APPROVED** [review]

**Pull request merged**